modules/core/04-channel/keeper/upgrade.go

```diff
@@ -123,38 +123,41 @@ func (k Keeper) ChanUpgradeTry(
 	}
 
 	var (
-		err             error
-		upgrade         types.Upgrade
-		isCrossingHello bool
+		err                     error
+		upgrade                 types.Upgrade
+		isCrossingHello         bool
+		expectedUpgradeSequence uint64
 	)
 
 	upgrade, isCrossingHello = k.GetUpgrade(ctx, portID, channelID)
-	if counterpartyUpgradeSequence < channel.UpgradeSequence {
+	if isCrossingHello {
+		expectedUpgradeSequence = channel.UpgradeSequence
+	} else {
+		expectedUpgradeSequence = channel.UpgradeSequence + 1
+	}
+	if counterpartyUpgradeSequence < expectedUpgradeSequence {
 		// In this case, the counterparty upgrade is outdated. We want to force the counterparty
 		// to abort their upgrade and come back to sync with our own upgrade sequence.
-		var upgradeSequence uint64
-		if isCrossingHello {
-			// In the crossing hello case, we already have an upgrade but it is at a higher sequence than the counterparty.
-			// Thus, our upgrade should take priority. We force the counterparty to abort their upgrade by invalidating all counterparty
-			// upgrade attempts below our own sequence by setting errorReceipt to upgradeSequence - 1.
-			// The upgrade handshake may then proceed on the counterparty with our sequence
-			upgradeSequence = channel.UpgradeSequence - 1
-		} else {
-			// In the case, that we are in a non-crossing hello (i.e. upgrade does not exist on our side),
-			// the sequence on both sides should move to a fresh sequence on the next upgrade attempt.
-			// Thus, we write an error receipt with our own upgrade sequence which will cause the counterparty
-			// to cancel their upgrade and move to the same sequence. When a new upgrade attempt is started from either
-			// side, it will be a fresh sequence for both sides (i.e. channel.upgradeSequence + 1)
-			upgradeSequence = channel.UpgradeSequence
-		}
+
+		// In the crossing hello case, we already have an upgrade but it is at a higher sequence than the counterparty.
+		// Thus, our upgrade should take priority. We force the counterparty to abort their upgrade by invalidating all counterparty
+		// upgrade attempts below our own sequence by setting errorReceipt to upgradeSequence - 1.
+		// Since our channel upgrade sequence was incremented in the previous step, the counterparty will be forced to abort their upgrade
+		// and we will be able to proceed with our own upgrade.
+		// The upgrade handshake may then proceed on the counterparty with our sequence
+		// In the case, that we are in a non-crossing hello (i.e. upgrade does not exist on our side),
+		// the sequence on both sides should move to a fresh sequence on the next upgrade attempt.
+		// Thus, we write an error receipt with our own upgrade sequence which will cause the counterparty
+		// to cancel their upgrade and move to the same sequence. When a new upgrade attempt is started from either
+		// side, it will be a fresh sequence for both sides (i.e. channel.upgradeSequence + 1)
```

Comment:
```suggestion
		// In the non-crossing hello (i.e. upgrade does not exist on our side),
		// the sequence on both sides should move to a fresh sequence on the next upgrade attempt.
		// Thus, we write an error receipt with our own upgrade sequence which will cause the counterparty
		// to cancel their upgrade and move to the same sequence. When a new upgrade attempt is started from either
		// side, it will be a fresh sequence for both sides (i.e. channel.upgradeSequence + 1). 
		// Note that expectedUpgradeSequence - 1 == channel.UpgradeSequence in the non-crossing hello case.
```